benchmarks/linear_algebra/blas/level3/symm/symm_generator.cpp

```diff
@@ -0,0 +1,160 @@
+	
+
+	computation add_all("[N, M]->{add_all[i,j]: 0<=i<N and 0<=j<=M }", expr(), true, p_float64, &symm);
+
+	
+
+	mat_mul_a_b.set_expression(expr(mat_mul_a_b(i, j, k-1) + A(k, i) * B(i, j)));
+	mult_alpha.set_expression(expr(alpha(0) * mat_mul_a_b(i, j , j-1))); 
+	mat_mul_b_a.set_expression(expr(mat_mul_b_a(i, j, k-1) + B(k, j) * A(k, i)));
+	mat_mul_a_b_e.set_expression(expr(A(i, i) * B(i, j)));
+	mult_alpha_k.set_expression(expr(alpha(0) * mat_mul_a_b_e(i, j))); 
+	mult_beta.set_expression(expr(beta(0) *  C_1(i,j, j-1))); 
+	add_all.set_expression(expr(add_all(i,j-1) + mat_mul_a_b_e(i, j) + mult_alpha(i,j, j-1) + mult_beta(i,j))); 
+
+
+	// -------------------------------------------------------
+	// Layer II
+	// -------------------------------------------------------
+	add_all.after(mult_beta, j); 
+	mult_beta.after(mult_alpha_k, j); 
+	mult_alpha_k.after(mat_mul_a_b_e, j ); 
+	mat_mul_a_b_e.after(mat_mul_b_a, j); 
+	mat_mul_b_a.after(mult_alpha, k); 
+	mult_alpha.after(mat_mul_a_b,k);
+	mat_mul_a_b.after(init, j);
+
+	// -------------------------------------------------------
+	// Layer III
+	// -------------------------------------------------------
+
+	buffer buf_SIZES("buf_SIZES", {2}, tiramisu::p_int32, a_input, &symm);
+	
+	buffer buf_A("buf_A", {N,N}, p_float64, a_input, &symm);
+	buffer buf_B("buf_B", {N,M}, p_float64, a_input, &symm);
+	buffer buf_C("buf_C", {N,M}, p_float64, a_input, &symm);
+	buffer buf_alpha("buf_alpha", {1}, p_float64, a_input, &symm);
+	buffer buf_beta("buf_beta", {1}, p_float64, a_input, &symm);
+
+
+	buffer buf_result("buf_result", {N,M}, p_float64, a_output, &symm);
+
+	//Store inputs
+	SIZES.set_access("{SIZES[e]->buf_SIZES[e]: 0<=e<2}");
+	
+	A.set_access("{A[i,k]->buf_A[i,k]}");
+	A.set_access("{A[i,i]->buf_A[i,i]}");
+	B.set_access("{B[k,j]->buf_B[k,j]}");
+	B.set_access("{B[i,j]->buf_B[i,j]}");
+	C_0.set_access("{C_0[i,j]->buf_C[i,j]}");
+	alpha.set_access("{alpha[0]->buf_alpha[0]}");
+	beta.set_access("{beta[0]->buf_beta[0]}");
+
+	add_all.set_access("{add_all[i,j]->buf_result[i,j]}");
+	C_1.set_access("{C_1[i,j,k]->buf_result[i,j]}");
+	mult_beta.set_access("{mult_beta[i,j]->buf_result[i,j]}");
+	mult_alpha_k.set_access("{mult_alpha_k[i,j]->buf_result[i,j]}");
+	mat_mul_a_b_e.set_access("{mat_mul_a_b_e[i,j]->buf_result[i,j]}");
+	mat_mul_b_a.set_access("{mat_mul_b_a[i,j,k]->buf_result[i,j]}");
+	mult_alpha.set_access("{mult_alpha[i,j,k]->buf_result[i,j]}");
+	mat_mul_a_b.set_access("{mat_mul_a_b[i,j, k]->buf_result[i,j]}");
+	init.set_access("{init[i,j]->buf_result[i,j]}");
+
+	
+	// -------------------------------------------------------
+	// Code Generation
+	// -------------------------------------------------------
+	symm.set_arguments({&buf_SIZES, &buf_A, &buf_B, &buf_C, &buf_alpha, &buf_beta, &buf_result});
+	symm.gen_time_space_domain();
+	symm.gen_isl_ast();
+	symm.gen_halide_stmt();
+	symm.gen_halide_obj("generated_symm.o");
+	
+	return 0;
+
+}
```

Comment:
Thanks. The code looks good to me. Can you please use the new Tiramisu syntax rather than the old syntax? The new syntax is the syntax used in the first three tutorials? (it is simpler and cleaner).

Reply:
Hi, 
Is it about the code generation section ?

Reply:
Hi Hadjer,

The whole file symm_generator.cpp  is written using the old Tiramisu syntax. Please check the first few tutorials. The syntax used in those tutorials is the new Tiramisu syntax. Use that one instead of the old syntax.

Reply:
Updated.

Reply:
Great! Your updates look good to me. There still a small issue with coding style that I pointed out in a comment.